build fails on linux 4.14.71-v7+ #1145 SMP Fri Sep 21 15:38:35 BST 2018 armv7l GNU/Linux

Trying to build on a raspberry pi

I get 
```
make -f Makefile.Release
make[1]: Entering directory '/home/pi/jacktrip/builddir'
g++ -c -pipe -D__LINUX_ALSA__ -g -O2 -D__LINUX_ALSA__ -O2 -std=gnu++11 -D_REENTRANT -Wall -W -fPIC -DWAIRTOHUB -D__LINUX__ -DQT_NO_DEBUG -DQT_NETWORK_LIB -DQT_CORE_LIB -I../src -I. -I../faust-src-lair/stk -isystem /usr/local/include -I../faust-src-lair -isystem /usr/include/arm-linux-gnueabihf/qt5 -isystem /usr/include/arm-linux-gnueabihf/qt5/QtNetwork -isystem /usr/include/arm-linux-gnueabihf/qt5/QtCore -Irelease -I/usr/lib/arm-linux-gnueabihf/qt5/mkspecs/linux-g++ -o release/DataProtocol.o ../src/DataProtocol.cpp
In file included from ../src/AudioInterface.h:41:0,
                 from ../src/jacktrip_globals.h:41,
                 from ../src/DataProtocol.cpp:39:
../src/ProcessPlugin.h: In member function 'virtual char* ProcessPlugin::getName()':
../src/ProcessPlugin.h:69:52: error: must #include <typeinfo> before using typeid
       char* pluginName { const_cast<char*>(typeid(*this).name()) }; // get name of DERIVED class
                                                    ^~~~
Makefile.Release:410: recipe for target 'release/DataProtocol.o' failed
make[1]: *** [release/DataProtocol.o] Error 1
make[1]: Leaving directory '/home/pi/jacktrip/builddir'
Makefile:38: recipe for target 'release' failed
make: *** [release] Error 2
```

it builds fine on other RPis I have lying around, don't know what wrong here...

--- a/src/Compressor.h
+++ b/src/Compressor.h
@@ -132,6 +132,7 @@ class Compressor : public ProcessPlugin
     int getNumInputs() override { return (mNumChannels); }
     int getNumOutputs() override { return (mNumChannels); }
     void compute(int nframes, float** inputs, float** outputs) override;
+    const char* getName() const override { return "Compressor"; }
 
    private:
     float fs;

--- a/src/Limiter.h
+++ b/src/Limiter.h
@@ -125,6 +125,7 @@ class Limiter : public ProcessPlugin
     int getNumInputs() override { return (mNumChannels); }
     int getNumOutputs() override { return (mNumChannels); }
     void compute(int nframes, float** inputs, float** outputs) override;
+    const char* getName() const override { return "Limiter"; }
 
     void setWarningAmplitude(double wa)
     {  // setting to 0 turns off warnings

--- a/src/NetKS.h
+++ b/src/NetKS.h
@@ -133,6 +133,8 @@ class NetKS : public ProcessPlugin
         }
     }
 
+    const char* getName() const override { return "NetKS"; }
+
     //============================================================================
 };
 

--- a/src/ProcessPlugin.h
+++ b/src/ProcessPlugin.h
@@ -63,13 +63,7 @@ class ProcessPlugin : public QThread
 
     //virtual void buildUserInterface(UI* interface) = 0;
 
-    virtual char* getName()
-    {
-        char* pluginName{
-            const_cast<char*>(typeid(*this).name())};  // get name of DERIVED class
-        while (isdigit(*pluginName)) { pluginName++; }
-        return pluginName;
-    }
+    virtual const char* getName() const = 0; // get name of DERIVED class
 
     /** \brief Do proper Initialization of members and class instances. By default this
    * initializes the Sampling Frequency. If a class instance depends on the
@@ -79,8 +73,7 @@ class ProcessPlugin : public QThread
     {
         fSamplingFreq = samplingRate;
         if (verbose) {
-            char* derivedClassName = getName();
-            printf("%s: init(%d)\n", derivedClassName, samplingRate);
+            printf("%s: init(%d)\n", getName(), samplingRate);
         }
     }
     virtual bool getInited() { return inited; }

--- a/src/Reverb.h
+++ b/src/Reverb.h
@@ -146,7 +146,7 @@ class Reverb : public ProcessPlugin
     int getNumInputs() override { return (mNumInChannels); }
     int getNumOutputs() override { return (mNumOutChannels); }
     void compute(int nframes, float** inputs, float** outputs) override;
-
+    const char* getName() const override { return "Reverb"; }
    private:
     float fs;
     int mNumInChannels;

